balancer/leastrequest/balancer_test.go

```diff
@@ -451,18 +430,59 @@ func (s) TestLeastRequestPersistsCounts(t *testing.T) {
 		t.Fatalf("error in expected round robin: %v", err)
 	}
 
-	for i := 0; i < 50; i++ {
-		var peer peer.Peer
-		stream, err := testServiceClient.FullDuplexCall(ctx, grpc.Peer(&peer))
+	// Start 25 rpcs, but don't finish them. They should all start on address 3,
+	// since the first two addresses both have 25 RPCs (and randomness
+	// injection/choiceCount causes all 3 to be compared every iteration).
+	for i := 0; i < 25; i++ {
+		stream, err := testServiceClient.FullDuplexCall(ctx)
 		if err != nil {
 			t.Fatalf("testServiceClient.FullDuplexCall failed: %v", err)
 		}
-		stream.CloseSend() // Finish stream to populate peer.
-		if _, err = stream.Recv(); err != io.EOF {
-			t.Fatalf("unexpected error: %v, expected an EOF error", err)
+		defer func() {
+			stream.CloseSend()
+			if _, err = stream.Recv(); err != io.EOF {
+				t.Fatalf("unexpected error: %v, expected an EOF error", err)
+			}
+		}()
+		p, ok := peer.FromContext(stream.Context())
+		if !ok {
+			t.Fatalf("testServiceClient.FullDuplexCall has no Peer")
 		}
-		if peer.Addr.String() != addresses[2] {
-			t.Fatalf("got: %v, want: %v", peer.Addr.String(), addresses[2])
+		if p.Addr.String() != addresses[2] {
+			t.Fatalf("testServiceClient.FullDuplexCall's Peer got: %v, want: %v", p.Addr.String(), addresses[2])
 		}
 	}
+
+	// Now 25 RPC's are active on each address, the next three RPC's should
+	// round robin, since choiceCount is three and the injected random indexes
+	// cause it to search all three addresses for fewest outstanding requests on
+	// each iteration.
+	wantAddrCount := map[string]int{
```

Comment:
For all of this and below, can you not use `checkRoundRobin`?

Reply:
No :/. checkRoundRobin gives it slack of it performs the operation within the ctx timeout. This is a lot more specific, as the moment address 3 gets populated with 25 active RPC's, *the next three RPC's should immediately be round robin*, not within the context of 5 seconds, three iterations of the three addresses happen.